source/gmsh/utilities.cc

```diff
@@ -144,15 +155,201 @@ namespace Gmsh
   }
 #  endif
 
+#  ifdef DEAL_II_GMSH_WITH_API
+  template <int dim, int spacedim>
+  void
+  read_partitioned_msh(
+    parallel::fullydistributed::Triangulation<dim, spacedim> &tria,
+    const MPI_Comm                                            mpi_comm,
+    const std::string                                        &file_prefix,
+    const std::string                                        &file_suffix)
+  {
+    const unsigned int nprocs = Utilities::MPI::n_mpi_processes(mpi_comm);
+    const unsigned int rank   = Utilities::MPI::this_mpi_process(mpi_comm);
+
+    std::string fname =
+      file_prefix + "_" + std::to_string(rank + 1) + "." + file_suffix;
+
+    if (nprocs == 1)
+      {
+        fname = file_prefix + "." + file_suffix;
+        std::ifstream f(fname);
+        AssertThrow(f.good(), ExcMessage("Missing mesh file: " + fname));
+      }
+    else
+      {
+        for (unsigned int i = 1; i <= nprocs; ++i)
+          {
+            const std::string check_fname =
+              file_prefix + "_" + std::to_string(i) + "." + file_suffix;
+            std::ifstream f(check_fname);
+            AssertThrow(f.good(),
+                        ExcMessage("Missing mesh file: " + check_fname));
+          }
+
+        const std::string extra_fname =
+          file_prefix + "_" + std::to_string(nprocs + 1) + "." + file_suffix;
+        std::ifstream f(extra_fname);
+        AssertThrow(!f.good(),
+                    ExcMessage("Unexpected extra mesh file(s): " +
+                               extra_fname));
```

Comment:
And here. I must admit that I don't know what the error message means. Are you testing that there are *exactly* $N$ files? If so, perhaps say in the error message that the function expected N files but also found an (N+1)st file.